src/services/issuer.rs

```diff
@@ -712,14 +713,70 @@ pub fn create_credential(
             cred_def, secret!(&cred_def_private), &cred_offer.nonce, &cred_request, secret!(&cred_values), revocation_config,
             );
 
-    let cred_public_key: anoncreds_clsignatures::CredentialPublicKey =
-        cred_def.get_public_key().map_err(err_map!(
+    let (credential_signature, signature_correctness_proof, rev_reg_id, rev_reg, witness) =
+        CLCredentialIssuer::init(cred_def, cred_def_private)?.create_credential(
+            cred_offer,
+            cred_request,
+            &cred_values,
+            revocation_config,
+        )?;
+
+    let credential = Credential {
+        schema_id: cred_offer.schema_id.clone(),
+        cred_def_id: cred_offer.cred_def_id.clone(),
+        rev_reg_id,
+        values: cred_values,
+        signature: credential_signature,
+        signature_correctness_proof,
+        rev_reg,
+        witness,
+    };
+
+    trace!(
+        "create_credential <<< credential {:?}",
+        secret!(&credential),
+    );
+
+    Ok(credential)
+}
+
+pub(crate) struct CLCredentialIssuer<'a> {
+    cred_def: &'a CredentialDefinition,
+    cred_def_private: &'a CredentialDefinitionPrivate,
+}
+
+impl<'a> CLCredentialIssuer<'a> {
+    pub(crate) fn init(
```

Comment:
Is there a specific reason why this is not just `new` and why does it return a `Result`?

Reply:
Can be updated. 
`CLCredentialIssuer` returns `Result` only to be aligned with other `CL` helpers (like `CLCredentialProver`) which does more steps on the `init` and truly returns Result.